File: eecs268/2020fall/12pm/2020.09.07.interfaces/Circle.cpp
//Circle.cpp
#include "Circle.h"

Circle::Circle(double radius)
{
	m_radius = radius;
}

double Circle::circumference() const
{
	return(2*3.14*m_radius);
}

//Satifies the requirement from the 
//Shape class
double Circle::area() const
{
	return(3.14*m_radius*m_radius);
}

File: eecs168/2022spring/TR/2022.04.19.classes-part-2/circle.py
#circle.py

class Circle:
    def __init__(self, radius=0):
        if radius > 0:
            self._radius = radius
        else:
            self._radius = 0
        #we don't normally do this
        print('You made a Circle')

    def area(self):
        ans = 3.14*self._radius**2
        return ans

    def circumference(self):
        ans = 2*3.14*self._radius
        return ans

    def set_radius(self, radius):
        if radius > 0:
            self._radius = radius
        else:
            self._radius = 0

    def get_radius(self):
        return self._radius

    def __lt__(self, other):
        if self._radius < other._radius:
            return True
        else:
            return False
